site/source/docs/api_reference/proxying.h.rst

```diff
@@ -0,0 +1,144 @@
+.. _proxying-h:
+
+==========
+proxying.h
+==========
+
+The Emscripten `proxying.h
+<https://github.com/emscripten-core/emscripten/blob/main/system/include/emscripten/proxying.h>`_
+API provides a mechanism for dispatching work to be executed by a target thread
+and optionally blocking until that work has been completed. This is particularly
+useful in a Web context where certain JS APIs can only be called on particular
+threads; any thread can access those APIs by proxying the API calls to the
+correct threads. Proxying is also useful for wrapping asynchronous JS tasks with
+a synchronous interface. The calling thread can proxy the asynchronous task to a
+dedicated worker thread and wait until the worker thread eventually marks the
+task as complete, possibly after returning multiple times to the JS event loop.
+
+Work on proxying queues can be executed in two ways. First, the user may
+explicitly call ``emscripten_proxy_execute_queue`` to execute any tasks that
+have been queued for the current thread. Alternatively, if the target thread
+returns to the JS event loop, for example via
+``emscripten_exit_with_live_runtime``, the proxying queue implementation will
+``postMessage`` the target thread to wake it up and tell it to automatically run
+the enqueued work.
```

Comment:
This paragraph seems a little unclear to me, but maybe I am missing something.

First, the user of the API can explicitly execute anything in the queue. Second, there are two things that cause execution to happen, either such an explicit call by the user, or if that thread returns to the main event loop then it will happen automatically. Correct?

Second, that latter case is implemented using a `postMessage` but I think that's an internal detail that might not be worth mentioning here, or maybe just mentioning in parentheses, as the user doesn't need to know about it.

What I think is maybe confusing me in the existing text is that it talks about what the user may do on a thread to run code there, then it switches to "target thread" which is a different perspective, from another thread sending to it.

Reply:
> This paragraph seems a little unclear to me, but maybe I am missing something.
> 
> First, the user of the API can explicitly execute anything in the queue. Second, there are two things that cause execution to happen, either such an explicit call by the user, or if that thread returns to the main event loop then it will happen automatically. Correct?

Correct!

> Second, that latter case is implemented using a `postMessage` but I think that's an internal detail that might not be worth mentioning here, or maybe just mentioning in parentheses, as the user doesn't need to know about it.

I guess the important part for the user is that they have to return to the JS event loop for this automatic execution to happen. I'll update the text.

> What I think is maybe confusing me in the existing text is that it talks about what the user may do on a thread to run code there, then it switches to "target thread" which is a different perspective, from another thread sending to it.

I will clarify this transition.